domain/application/state/application_test.go

```diff
@@ -902,6 +902,81 @@ func (s *applicationStateSuite) TestStorageDefaults(c *gc.C) {
 	})
 }
 
+func (s *applicationStateSuite) TestCreateApplicationThenGetCharmIDByApplicationName(c *gc.C) {
```

Comment:
Is this really just to test `GetCharmIdByApplicationName`? I think its reasonable to just call it `TestGetCharmIdByApplicationName`, the creating of the application can be viewed as test setup, it would be pretty clear for someone reading the test why it was done.